test/parallel/test-process-setuid-setgid.js

```diff
@@ -20,25 +20,49 @@
 // USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 'use strict';
-// Requires special privileges
 const common = require('../common');
+
 const assert = require('assert');
 
-var oldgid = process.getgid();
-process.setgid('nobody');
-var newgid = process.getgid();
-assert.notStrictEqual(newgid, oldgid, 'gids expected to be different');
+if (common.isWindows) {
+  // uid/gid functions are POSIX only
+  assert.strictEqual(process.getuid, undefined);
+  assert.strictEqual(process.setuid, undefined);
+  assert.strictEqual(process.getgid, undefined);
+  assert.strictEqual(process.setgid, undefined);
+  return;
+}
 
-var olduid = process.getuid();
-process.setuid('nobody');
-var newuid = process.getuid();
-assert.notStrictEqual(newuid, olduid, 'uids expected to be different');
-
-try {
-  process.setuid('nobody1234');
-} catch (e) {
-  assert.strictEqual(e.message,
-               'failed to resolve group',
-               'unexpected error message'
+assert.throws(() => {
+  process.setuid('fhqwhgadshgnsdhjsdbkhsdabkfabkveybvf');
+}, /^Error: setuid user id does not exist$/);
+
+// If we're not running as super user...
+if (process.getuid() !== 0) {
+  assert.doesNotThrow(() => {
+    process.getgid();
+    process.getuid();
+  });
+
+  assert.throws(
+    () => { process.setgid('nobody'); },
+    /^Error: (EPERM, .+|setgid group id does not exist)$/
```

Comment:
> Also, the capture group doesn't look correct to me.

@thefourtheye There's room for improvement. For one thing, it could be a non-capturing group.  But the regexp seems to work to me. What situation are you concerned about?

```console
> /^Error: (EPERM, .+|setgid group id does not exist)$/.test('Error: setgid group id does not exist')
true
> /^Error: (EPERM, .+|setgid group id does not exist)$/.test('Error: setgid group id does not exist adding some text so it will be false')
false
> /^Error: (EPERM, .+|setgid group id does not exist)$/.test('Error: EPERM, setgid group id does not exist adding some text so it will be false except now it as EPERM at the start so it should be true again')
true
> 
```